pallets/marketplace/src/lib.rs

```diff
@@ -383,6 +385,7 @@ pub mod pallet {
             origin: OriginFor<T>, // source
             job_id: JobId<T::AccountId>,
             last: bool,
+            failure_message: Option<ExecutionFailureMessage>,
```

Comment:
We should accept `Successful` and `Faulty` messages. In case of success, we expect the operation hash.

Reply:
updated